backend/src/core/logs/providers/cloudwatch.provider.ts

```diff
@@ -290,106 +312,23 @@ export class CloudWatchProvider extends BaseAnalyticsProvider {
       let lastActivity = '';
 
       if (sourceStreams.length > 0) {
-        try {
-          // Use EXACTLY the same approach as getLogsBySource to get consistent results
-          const endMs = Date.now();
-          const startMs = endMs - 24 * 60 * 60 * 1000; // Look back 24 hours (same as getLogsBySource)
-
-          // Use CloudWatch Insights to efficiently get the latest timestamp
-          try {
-            const end = Date.now();
-            const start = end - 24 * 60 * 60 * 1000; // Last 24 hours
-
-            const insights = `fields @timestamp | filter @logStream like /${suffix}/ | sort @timestamp desc | limit 1`;
-
-            const startQuery = await client.send(
-              new StartQueryCommand({
-                logGroupName: logGroup,
-                startTime: Math.floor(start / 1000),
-                endTime: Math.floor(end / 1000),
-                queryString: insights,
-                limit: 1,
-              })
-            );
-
-            const qid = startQuery.queryId || '';
-            const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms));
-            let results;
-
-            // Wait for query to complete (shorter timeout for stats)
-            for (let i = 0; i < 10; i++) {
-              const r = await client.send(new GetQueryResultsCommand({ queryId: qid }));
-              if (r.status === 'Complete' || r.status === 'Failed' || r.status === 'Cancelled') {
-                results = r.results || [];
-                break;
-              }
-              await sleep(200);
+        // Use stream metadata for last activity (most reliable and fastest)
+        const last = streams
```

Comment:
Based on my previous testing, the streaming method can cause log loss, especially for the latest log. Since the logs have already been paged out, the returned results are incomplete. You can test this method using Pokemon Vibe, starting with the latest log and paged out five times in a row to see if the logs returned by this API correspond exactly to the actual logs, ensuring that no logs are lost.
If this test passes, then there's no problem.

Reply:
There is no problem in getting the log once, but to ensure that the log is continuous and not lost, testing and verification are still needed.